### PR TITLE
fix/5415-PR-creation-issue

### PR DESCRIPTION
**Pull request by mihailShumilov**
Close: #5415

## PR Details

If Billing is disabled you can't create a PR

## PR Checklist

- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-server/src/core/build/build.service.ts [hunk 1]
```diff
@@ -454,15 +454,18 @@ export class BuildService {
...
-          const smartGitSyncEntitlement =
-            await this.billingService.getBooleanEntitlement(
-              project.workspaceId,
-              BillingFeature.SmartGitSync
-            );
-
-          const pullRequestMode = smartGitSyncEntitlement.hasAccess
-            ? EnumPullRequestMode.Accumulative
-            : EnumPullRequestMode.Basic;
+          let pullRequestMode = EnumPullRequestMode.Accumulative;
```
Comment (tupe12334) [inline packages/amplication-server/src/core/build/build.service.ts:457]: I think the default suppose to be Basic, @EugeneTseitlin Please have a look here

Comment (mihailShumilov) [reply packages/amplication-server/src/core/build/build.service.ts:457]: usually billing is disabled for local env, so I think that we can use `Accumulative` mode
Or we can set that mode in `.env` variable

Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/build/build.service.ts:457]: > I think the default suppose to be Basic, @EugeneTseitlin Please have a look here

Sure, it has to be basic by default.

**Review by EugeneTseitlin — CHANGES_REQUESTED** [review]

File: packages/amplication-server/src/core/build/build.service.ts [hunk 2]
```diff
@@ -454,15 +454,18 @@ export class BuildService {
...
+            const smartGitSyncEntitlement =
+              await this.billingService.getBooleanEntitlement(
+                project.workspaceId,
+                BillingFeature.SmartGitSync
+              );
+
+            pullRequestMode = smartGitSyncEntitlement.hasAccess
+              ? EnumPullRequestMode.Accumulative
+              : EnumPullRequestMode.Basic;
+          }
```
Comment (abrl91) [inline packages/amplication-server/src/core/build/build.service.ts:468]:
```suggestion
          const smartGitSyncEntitlement = this.billingService.isBillingEnabled ? await this.billingService.getBooleanEntitlement(
                project.workspaceId,
                BillingFeature.SmartGitSync
              ) : false
          
```

Comment (abrl91) [reply packages/amplication-server/src/core/build/build.service.ts:468]: and in line 490:
```ts
pullRequestMode: smartGitSyncEntitlement?.hasAccess 
             ? EnumPullRequestMode.Accumulative
              : EnumPullRequestMode.Basic;
```

File: packages/amplication-server/src/core/build/build.service.ts [hunk 3]
```diff
@@ -484,7 +481,11 @@ export class BuildService {
               adminUIPath: resourceInfo.settings.adminUISettings.adminUIPath,
               serverPath: resourceInfo.settings.serverSettings.serverPath,
             },
-            pullRequestMode,
+            pullRequestMode:
```
Comment (abrl91) [inline packages/amplication-server/src/core/build/build.service.ts:484]: What is the value of `pullRequestMode` if `smartGitSyncEntitlement` returns `false` (meaning typeof boolean)?

Comment (abrl91) [reply packages/amplication-server/src/core/build/build.service.ts:484]: I think you can do something like that:
declare the pull request mode
```suggestion
          const pullRequestMode =
            (smartGitSyncEntitlement &&
              smartGitSyncEntitlement.hasAccess &&
              EnumPullRequestMode.Accumulative) ||
            EnumPullRequestMode.Basic; 
```
and then use it here

File: packages/amplication-server/src/core/build/build.service.ts [hunk 4]
```diff
@@ -484,7 +481,11 @@ export class BuildService {
...
+              typeof smartGitSyncEntitlement !== "boolean" &&
+              smartGitSyncEntitlement?.hasAccess
+                ? EnumPullRequestMode.Accumulative
+                : EnumPullRequestMode.Basic,
```
Comment (abrl91) [inline packages/amplication-server/src/core/build/build.service.ts:488]: I think you can do something like that:
```suggestion
          pullRequestMode: (smartGitSyncEntitlement &&
              smartGitSyncEntitlement.hasAccess &&
              EnumPullRequestMode.Accumulative) ||
            EnumPullRequestMode.Basic; 
```
or declare it above in a const and use the constant here

File: packages/amplication-server/src/core/build/build.service.ts [hunk 5]
```diff
@@ -484,7 +481,10 @@ export class BuildService {
               adminUIPath: resourceInfo.settings.adminUISettings.adminUIPath,
               serverPath: resourceInfo.settings.serverSettings.serverPath,
             },
-            pullRequestMode,
+          pullRequestMode: (smartGitSyncEntitlement &&
+              smartGitSyncEntitlement.hasAccess &&
+              EnumPullRequestMode.Accumulative) ||
+            EnumPullRequestMode.Basic;
```
Comment (tupe12334) [inline packages/amplication-server/src/core/build/build.service.ts:487]:
```suggestion
          pullRequestMode: (smartGitSyncEntitlement &&
              smartGitSyncEntitlement.hasAccess) ?
              EnumPullRequestMode.Accumulative :
            EnumPullRequestMode.Basic; 
```

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**